.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+# Notes:
+#  - All software installed below is pinned so they don't get upgraded and break w/o us knowing why.
+#  - If you bump one version up to latest, please bump them all so we stay current!
+
+name: Build couch2pg and run test cases
+
+on: [push, pull_request]
+
+jobs:
+  build-and-test:
+    name: Build couch2pg
+    runs-on: ubuntu-18.04
+    env:
+      INT_COUCHDB_URL: http://localhost:5984/medic-analytics-test
+      INT_PG_PASS: postgrespass
+      INT_PG_USER: postgresuser
+      INT_PG_DB: medic-analytics-test
+
+    services:
+      postgres:
+        image: postgres:9.4
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_PASSWORD: ${{ env.INT_PG_PASS }}
+          POSTGRES_USER: ${{ env.INT_PG_USER }}
+          POSTGRES_DB: ${{ env.INT_PG_DB }}
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Set up CouchDB
```

Comment:
Could this be moved to the services containers instead of using a third party github action?

Reply:
The 3rd party action does a few more steps that the docker image doesn't do by default, particularly creating the system dbs at the bottom of [the entrypoint script](https://github.com/cobot/couchdb-action/blob/master/entrypoint.sh). We could inline those but I'm happy using the third party action to save us maintaining those steps. What do you think?

Reply:
Nope I think that's fine. One last request would be to use the commit hash on third party actions. It is recommended by the docs. Also using straight master seems  bit concerning.  I think this warrants a change to our docs too.  I'll put  PR up.

https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions